app/models/piece.rb

```diff
@@ -89,19 +105,69 @@ def move_to!(x, y)
       return 'invalid move'
     end
     self.update_attributes(x_position: x, y_position: y)
+    self.increment!(:moves)
+    if castling_queenside? || castling_kingside?
+      handle_castling(y)
+    end
+  end
+
+  def handle_castling(y) #triggered once the king is moved to a castling spot
+    update_rook_if_castling(y)
+  end
+
+  def castling_queenside? #tests to see if king is castling towards kingside
+    type == "King" &&
+    x_position == 3 &&
+    self.moves == 1 &&
+    y_position == white? ? 8 : 0 &&
+    rook_at(1,y_position)
   end
 
+  def castling_kingside? #tests to see if king is castling towards kingside
+    type == "King" &&
+    x_position == 7 &&
+    self.moves == 1 &&
+    y_position == white? ? 8 : 0  &&
+    rook_at(8,y_position)
+  end
+
+  def update_rook_if_castling(y) #updates the rooks position when the king is moved to a castling spot
```

Comment:
This was always moving the king side rook because both castling_kingside? and castling_queenside? would return true no matter where you were moving the King to.